QualityControl/test/public/pages/layout-view.test.js

```diff
@@ -10,153 +10,223 @@
-      return {
-        lastModified: lastModified,
-        path: path,
-        title: infoButtonTitle,
-      };
-    });
-    assert.strictEqual(result.title, 'View details about histogram', 'Button title is different');
-    assert.ok(result.path.includes('PATH'), 'Object full path label is not the same');
-    assert.ok(result.path.includes('DAQ01/EventSizeClasses/class_C0ALSR-ABC'), 'Object full path is not the same');
-    assert.ok(result.lastModified.includes('LAST MODIFIED'), 'Last Modified label is different');
-  });
-
-  it('should have an info button with full path and last modified when clicked on a second plot(plot success)', async () => {
-    const result = await page.evaluate(() => {
-      const infoButtonTitle = document.querySelector('body > div > div > section > div > div > div:nth-child(2) > div > div > div:nth-child(2) > div > div > button').title;
-      const lastModified = document.querySelector('body > div > div > section > div > div > div:nth-child(2) > div > div > div:nth-child(2) > div > div > div > div:nth-child(2)').innerText;
-      const path = document.querySelector('body > div > div > section > div > div > div:nth-child(2) > div > div > div:nth-child(2) > div > div > div > div').innerText;
-      return {
-        lastModified: lastModified,
-        path: path,
-        title: infoButtonTitle,
-      };
+
+import { strictEqual, ok } from 'node:assert';
+import { delay } from '../../testUtils/delay.js';
+
+/**
+ * Performs a series of automated tests on the layoutShow page using Puppeteer.
+ * @param {string} url - URL needed to open page for testing
+ * @param {object} page - Puppeteer page object
+ * @param {number} timeout - Timeout PER test; default 100
+ * @param {object} testParent - Node.js test object which ensures sub-tests are being awaited
+ */
+export const layoutShowTests = async (url, page, timeout = 5000, testParent) => {
+  const LAYOUT_ID = '671b95883d23cd0d67bdc787';
+  await testParent.test(
+    'should load',
+    { timeout },
+    async () => {
+      await page.goto(`${url}?page=layoutShow&layoutId=${LAYOUT_ID}`, { waitUntil: 'networkidle0' });
+      const location = await page.evaluate(() => window.location);
+      strictEqual(location.search, `?page=layoutShow&layoutId=${LAYOUT_ID}&tab=main`);
+    },
+  );
+
+  await testParent.test(
+    'should have tabs in the header',
+    { timeout },
+    async () => {
+      const tabsPath = 'header .btn-tab';
+      const tabsCount = await page.evaluate((tabsPath) => document.querySelectorAll(tabsPath).length, tabsPath);
+      strictEqual(tabsCount, 2);
+    },
+  );
+
+  await testParent.test(
+    'should have selected layout in the sidebar highlighted',
+    { timeout },
+    async () => {
+      const layoutClassList = await page.evaluate(() => document
+        .querySelector('nav > div:nth-child(5) > a:nth-child(1)').classList);
+      strictEqual(
```

Comment:
I believe here you could make use of the `deepStrictEqual` method instead of `strictEqual`. This will allow you to not have to use `JSON.stringify` anymore